onchain/src/templates/component.cairo

```diff
@@ -59,11 +62,38 @@ pub mod TemplateStoreComponent {
             let template_id = self.templates_count.read();
             self.templates.write(template_id, template_metadata);
             self.templates_count.write(template_id + 1);
+            self
+                .deposit(
+                    get_caller_address(), template_metadata.reward_token, template_metadata.reward
+                );
             self.emit(TemplateAdded { id: template_id, metadata: template_metadata });
         }
 
         fn is_template_complete(self: @ComponentState<TContractState>, template_id: u32) -> bool {
             self.completed_templates.read(template_id)
         }
     }
+
+    #[generate_trait]
+    impl InternalImpl<
+        TContractState, +HasComponent<TContractState>
+    > of InternalTrait<TContractState> {
+        fn deposit(
+            ref self: ComponentState<TContractState>,
+            template_proposer: ContractAddress,
```

Comment:
`template_proposer` arg can be removed and just use the `caller_address` like you did in the function.

Reply:
oh yeah true, my bad. thanks!